File: java/hello/NameSorter.java
import java.util.*;

public class NameSorter {
    public int compare(String s1, String s2) {
        return Integer.compare(s1.length(), s2.length());
    }
    public static void main(final String... args) {
        String[] names = { "Kumar", "Nikude", "Jain", "Bastade", "Koot" };
        Arrays.sort(names, NameSorter::compare);
        Arrays.sort(names, (s1, s2) -> s1.length() - s2.length());
        for (var name : names)
            System.out.println(name);
    }
}

File: kotlin/hello/NameSorter.kt
fun compareByLength(s1: String, s2: String) = s1.length - s2.length

fun main() {
    val names = arrayOf("Kumar", "Nikude", "Jain", "Bastade", "Koot")

    names.sortWith(::compareByLength)
    names.sortWith { s1, s2 -> s1.length - s2.length }
    names.sortBy { it.length }
    for (name in names)
        println(name)
}
